### PR TITLE
Added Fresnel integrals #398

### PR DESCRIPTION
**Pull request by curio-sitas**
Added Fresnel integrals from https://github.com/kiranshila/FresnelIntegrals.jl. As suggested in #398

Added 3 functions with docs : 
- Fresnelc(z)
- Fresnels(z)
- Fresnnel(z) = (Fresnelc(z),Fresnels(z))

Tests were added too

### CODE REVIEW

**Comment by curio-sitas** [comment]
Added the last test for the function fresnel(z::Number) for full code coverage (codecov)

File: src/fresnel.jl [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+"""
+        fresnelc(z)
+Calculates the Fresnel cosine integral for the number z for
+    ``C(z) = \\int_{0}^{z} \\cos{\\left(\\frac{\\pi t^2}{2}\\right)}dt``
+"""
+fresnelc(z::Number) = 0.25*(1-1im)*(1im*erf(0.5*(1-1im)*z*√(π)) + erf(0.5*(1+1im)*z*√(π)))
+
+"""
+        fresnels(z)
```
[the remark this answers is not in the document]
Comment (devmotion) [reply src/fresnel.jl:14]: Again, maybe rename to `fresnelsin`?

File: src/fresnel.jl [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+Calculates the Fresnel sine integral for the number z for
+    ``S(z) = \\int_{0}^{z} \\sin{\\left(\\frac{\\pi t^2}{2}\\right)}dt``
+"""
+fresnels(z::Number) = 0.25*(1+1im)*(-1im*erf(0.5*(1-1im)*z*√(π)) + erf(0.5*(1+1im)*z*√(π)))
```
Comment (devmotion) [inline src/fresnel.jl:18]: Similar to above:
```suggestion
function fresnels(z::Number)
    x = (z * sqrtπ) / 2
    re_x, im_x = reim(x)
    a = (re_x - im_x) + (re_x - im_x) * im
    b = (re_x + im_x) + (im_x - re_x) * im
    re_erf_a, im_erf_a = reim(erf(a))
    re_erf_b, im_erf_b = reim(erf(b))
    re_y = (re_erf_a - im_erf_a + re_erf_b + im_erf_b) / 4
    im_y = (re_erf_a + im_erf_a - re_erf_b + im_erf_b) / 4
    y = re_y + im_y * im
    return y
end
```

**Review by devmotion — COMMENTED** [review]
Thanks for the PR!

Unfortunately, I'm not sure if it will be merged since we have been hesitant with adding functionality to SpecialFunctions that already exists in other packages. Also here one could just load FresnelIntegrals, and it seems it's not necessary to include these functions in SpecialFunctions.

Nevertheless, I added comments and suggested some improvements. Even if this PR is not merged it might be useful to contribute those to FresnelIntegrals.

**Comment by curio-sitas** [comment]
Thanks for the review. I used to think it was a good idea to enrich the current package since it is a global SpecialFunctions collection.

I will take your reviews into account in my next commit and may open a PR on the original repo.

**Comment by curio-sitas** [comment]
- Sign errors were corrected and some tests added (comparison with QuadGK.jl).

All above is also proposed in a [PR on the original repo](https://github.com/kiranshila/FresnelIntegrals.jl/pull/2) (FresnelIntegrals.jl).

**Comment by devmotion** [comment]
@curio-sitas I guess this PR can be closed now since the PR in FresnelIntegrals.jl was merged and released in version 0.2?

**Comment by curio-sitas** [comment]
Yes you can close it !

**Comment by Bumblebee00** [comment]
> we have been hesitant with adding functionality to SpecialFunctions that already exists in other packages
can I ask why? Isnt a good idea to have the package SpecialFunctions.jl to contain a lot of special functions?

**Pull request closed without merging**